### DISCUSSION TITLE
Switch flag to ContinueOnError

### DISCUSSION THREAD

**Discussion by xiu** [prometheus-operator/prometheus-operator#7152] [Ideas]
When upgrading kube-prometheus-stack, I've been hit by the operator not wanting to start because of a flag not being defined:
`flag provided but not defined: -kubelet-endpoints`

This is due to the FlagSet being set with `flag.ExistOnError` in https://github.com/prometheus-operator/prometheus-operator/blob/main/cmd/operator/main.go#L660 (flag.CommandLine sets ErrorHandling to ExitOnError). Changing this the following should allow for the operator to continue even if it finds undefined flags:
```go
	os.Exit(run(flag.NewFlagSet(os.Args[0], flag.ContinueOnError))
```

My issue is more of a question, should the operator die if it finds a an undefined flag?

## Steps to Reproduce
Deploy prometheus-operator version <0.77.0 using `kube-prometheus-stack` `66.1.1`.

## Expected Result
The operator ignores the flag.

## Actual Result
The operator crashes.

**Comment by simonpasquier** [comment]
My personal preference is to break hard on unknown arguments because it can be tricky to find out that you had a typo in your args list.

**Reply by xiu** [reply]
Fair point, and a check could be done in projects using the operator to make sure this doesn't happen.
